pkg/api/soc_test.go

```diff
@@ -0,0 +1,165 @@
+			Tags:   tag,
+		})
+	)
+
+	t.Run("cmpty data", func(t *testing.T) {
+		jsonhttptest.Request(t, client, http.MethodPost, socResource("8d3766440f0d7b949a5e32995d09619a7f86e632", "bb", "cc"), http.StatusBadRequest,
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "short chunk data",
+				Code:    http.StatusBadRequest,
+			}),
+		)
+	})
+
+	t.Run("malformed id", func(t *testing.T) {
+		jsonhttptest.Request(t, client, http.MethodPost, socResource("8d3766440f0d7b949a5e32995d09619a7f86e632", "bbzz", "cc"), http.StatusBadRequest,
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "bad id",
+				Code:    http.StatusBadRequest,
+			}),
+		)
+	})
+
+	t.Run("malformed owner", func(t *testing.T) {
+		jsonhttptest.Request(t, client, http.MethodPost, socResource("xyz", "aa", "bb"), http.StatusBadRequest,
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "bad owner",
+				Code:    http.StatusBadRequest,
+			}),
+		)
+	})
+
+	t.Run("malformed signature", func(t *testing.T) {
+		jsonhttptest.Request(t, client, http.MethodPost, socResource("8d3766440f0d7b949a5e32995d09619a7f86e632", "aa", "bad sig"), http.StatusBadRequest,
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "bad signature",
+				Code:    http.StatusBadRequest,
+			}),
+		)
+	})
+
+	t.Run("signature invalid", func(t *testing.T) {
+		s, owner, payload := mockSoc(t)
+		id := make([]byte, soc.IdSize)
+
+		// modify the sign
+		sig := s.Signature()
+		sig[12] = 0x98
+		sig[10] = 0x12
+
+		jsonhttptest.Request(t, client, http.MethodPost, socResource(hex.EncodeToString(owner), hex.EncodeToString(id), hex.EncodeToString(sig)), http.StatusBadRequest,
+			jsonhttptest.WithRequestBody(bytes.NewReader(payload)),
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "invalid chunk",
+				Code:    http.StatusBadRequest,
+			}),
+		)
+	})
+
+	t.Run("ok", func(t *testing.T) {
+		s, owner, payload := mockSoc(t)
+		id := make([]byte, soc.IdSize)
+
+		sig := s.Signature()
+		addr, err := s.Address()
+		if err != nil {
+			t.Fatal(err)
+		}
+		jsonhttptest.Request(t, client, http.MethodPost, socResource(hex.EncodeToString(owner), hex.EncodeToString(id), hex.EncodeToString(sig)), http.StatusOK,
+			jsonhttptest.WithRequestBody(bytes.NewReader(payload)),
+			jsonhttptest.WithExpectedJSONResponse(api.SocPostResponse{
+				Reference: addr,
+			}),
+		)
+
+		// try to fetch the same chunk
+		rsrc := fmt.Sprintf("/chunks/" + addr.String())
+		resp := request(t, client, http.MethodGet, rsrc, nil, http.StatusOK)
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ch, err := s.ToChunk()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !bytes.Equal(ch.Data(), data) {
+			t.Fatal("data retrieved doesnt match uploaded content")
+		}
+	})
+}
+
+// returns a valid, mocked SOC
+func mockSoc(t *testing.T) (*soc.Soc, []byte, []byte) {
```

Comment:
why do we need this, use the soc pkg constructors, no need to reimplement it here

Reply:
they dont provide the necessary functionality